cmd/autotestgridgenerator/main.go

```diff
@@ -0,0 +1,184 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/experiment/autobumper/bumper"
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/labels"
+)
+
+const (
+	githubOrg    = "kubernetes"
+	githubRepo   = "test-infra"
+	githubLogin  = "openshift-bot"
+	githubTeam   = "openshift/openshift-team-developer-productivity-test-platform"
+	matchTitle   = "Automate testgrid generator"
+	remoteBranch = "auto-testgrid-generator"
+)
+
+type options struct {
+	selfApprove bool
```

Comment:
Do we plan to convince someone to give write rights to `k/test-infra`? (I guess that's more a question for @stevekuznetsov). Because if not, then we do not need this code.

Reply:
Currently we dont plan on that, but I left it in there in case we wanted to in the future. I'll remove it for now though